packages/zowe-explorer/src/shared/context.ts

```diff
@@ -386,10 +386,14 @@ export function asFavorite(node: TreeItem): string {
 }
 
 export function withProfile(node: IZoweTreeNode): string {
-    if (node?.contextValue?.includes(".profile=")) return node?.contextValue;
-    return node?.getParent()?.contextValue?.includes(".profile=")
-        ? node?.contextValue + ".profile=" + node?.getParent()?.contextValue?.split(".profile=")[1] ?? node?.contextValue
-        : node?.contextValue;
+    if (!node) return;
+    const hasProfile = (n: IZoweTreeNode) => n?.contextValue?.includes(".profile=") ?? false;
+    if (hasProfile(node)) return node.contextValue;
+    if (hasProfile(node.getParent())) {
```

Comment:
This *might* be optimized by JavaScript at some point, but I see there are two calls to `getParent` here. If this will increase code complexity in SonarCloud by declaring the parent, then this is good as-is 👍

Reply:
I kind of forgot about this one 😢 
I don't mind storing that in a variable even though it's just a getter for the `mParent` property
```typescript
public getParent(): IZoweTreeNode {
   return this.mParent;
}
```